### PR TITLE
Minimize dependency on Viper and make config read-only.

### PR DESCRIPTION
**Pull request by jeremyje**

### CODE REVIEW

**Comment by ihrankouski** [comment]
Maybe also it's a time to move config.go somewhere under /internal/ ?

**Comment by jeremyje** [comment]
I thought about that. Perhaps in a different PR. I don't want to have to
update a bunch of symlinks and the might be a better solution once the new
dockerfile go in.

File: Makefile [hunk 1]
```diff
@@ -70,6 +70,8 @@ EXE_EXTENSION =
 LOCAL_CLOUD_BUILD_PUSH = # --push
 KUBECTL_RUN_ENV = --env='REDIS_SERVICE_HOST=$$(OPEN_MATCH_REDIS_MASTER_SERVICE_HOST)' --env='REDIS_SERVICE_PORT=$$(OPEN_MATCH_REDIS_MASTER_SERVICE_PORT)'
 GCP_LOCATION_FLAG = --zone $(GCP_ZONE)
+# Flags to simulate behavior of newer versions of Kubernetes
```
[the remark this answers is not in the document]
Comment (jeremyje) [reply Makefile]: It's not it's probably a hold over form a different PR but I think it's ok

**Pull request merged**